tycho-bundles/org.eclipse.tycho.p2.resolver.impl.test/src/test/java/org/eclipse/tycho/p2/target/TestResolverFactory.java

```diff
@@ -104,4 +106,9 @@ public TargetPlatformFactoryImpl getTargetPlatformFactoryImpl() {
     public P2Resolver createResolver(MavenLogger logger) {
         return new P2ResolverImpl(getTargetPlatformFactoryImpl(), mavenContext.getLogger());
     }
+
+    @Override
```

Comment:
Couldn't this be a method directly in `ArtifactDescriptor` ? (it would probably require to move ArtifactDescriptor to org.eclipse.tycho.embedder.shared, but I imagine it would clarify things)

Reply:
The we need to move ArtifactKey as well. In general I think it would be good to merge all these "shared" bundles as this actually complicates/duplicates things a lot. 

Beside that, the structure is a bit split as the ArtifactDescriptor is created in the "maven" world one can't access the InstallableUnit there... it would really be good if we could make the P2 Api avaiable in the maven world but I think this would require a change in P2 to be more split between API, SPI and actual Implementation, something that most probably would not be compatible with platform API rules....